clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -686,6 +696,207 @@ check_function_markers()
         if (!run_checker(memrefs, false))
             return false;
     }
+    // Correctly handle signal arriving between a branch instruction and the function
+    // entry.
+    {
+        constexpr addr_t SIG_HANDLER_PC = 400;
+        constexpr addr_t SYSCALL_PC = 500;
+        constexpr size_t SYSCALL_SZ = 2;
+        constexpr size_t RETURN_SZ = 3;
+
+        std::vector<memref_t> memrefs = {
+            gen_instr(TID, 1),
+            gen_instr_type(TRACE_TYPE_INSTR_DIRECT_CALL, TID, CALL_PC, CALL_SZ),
+            gen_marker(TID, TRACE_MARKER_TYPE_KERNEL_EVENT, CALL_PC + CALL_SZ),
```

Comment:
The signal arrived at the PC right after the call?  Meaning the call targeted the very next instr?  That makes it confusing: normally that is only done on x86 when trying to get the retaddr, not when actually making a call, as for an actual call you can't then have a normal return from the callee.  I would request changing this to a realistic callee PC.